mesh/nurbs.hpp

```diff
@@ -262,6 +267,19 @@ class NURBSPatch
    NURBSPatch(const KnotVector *kv0, const KnotVector *kv1,
               const KnotVector *kv2, int dim);
 
+   /** Create a bivariate NURBS patch with given control points. See n-variate overload for additional notes. */
+   NURBSPatch(const KnotVector *kv0, const KnotVector *kv1, int dim_,
+              const double* control_points);
```

Comment:
In general, `real_t` should be used instead of `double` (MFEM supports `float` and `double`). Instead of a raw pointer, how about using `mfem::Vector`?

Reply:
I've changed all the `double`s to `real_t`s, but left `control_points` as a raw `real_t*` in this particular case, by analogy to other cases like sparse matrix constructors where the data pointed to doesn't have the direct interpretation of a 1D array.  (The knot vector constructors, on the other hand, I've updated to use `mfem::Vector`.)